File: py/btools.py
from telegram.client import Telegram
from chats import print_chats
from msgs import *

#tts language
LANG = "uk"

#phone_number = ""
phone_number = input("phone: ")

"""
                AUTH
"""

tg = Telegram(
    api_id=11280565,
    api_hash='5561e9f775bae9639412059f1cd4345d',
    phone = phone_number,
    use_message_database = False,
    database_encryption_key='',
)

tg.login()

run = True

while run:

    chat_list = tg.get_chats()
    chat_list.wait()

    print_chats(tg, chat_list.update["chat_ids"])
    chat_id = 0

    try:
        chat_id = int(input("select a chat: "))
    except ValueError:
        print("something is wrong with your integer, buddy")

    if chat_id in chat_list.update["chat_ids"]:
        
        chat_loop = True
        while chat_loop:

            user_inp = input("?: ")

            if len(user_inp) > 1:
                #process a command
                if user_inp[0] == "\\":
                    
                    if len(user_inp) > 3:

                        if user_inp[1] == "v":
                            send_ttm_voice_note(tg, chat_id, user_inp[3:], LANG)

                        elif user_inp[1] == "s":
                            send_file_voice_note(tg, chat_id, user_inp[3:])

                    #select another chat
                    elif user_inp[1] == "q": chat_loop = False
                    
                    #exit
                    if user_inp[1] == "e": 

                        chat_loop = False
                        run = False

                    else:
                        print("uknown command")
                else:
                    tg.send_message(chat_id = chat_id, text = user_inp)
            else:
                tg.send_message(chat_id = chat_id, text = user_inp)

    else:
        print("cant seem to find specified chat")


tg.stop()

File: go/btools.go
package main

import "os"
import "fmt"
import "bufio"
import "strconv"


var gLang string  = "uk"
var gChatLimit int32 = 15


func main() {

    //auth
    client := getClient() 
    defer client.Close()
    authLoop(client)
   
    var userInput string
    stdin := bufio.NewReader(os.Stdin)

    run := true

    for run {

        chats, err := getChats(client, gChatLimit)
        if err != nil { panic(err) }

        printChats(chats)

        fmt.Print("Select a chat: ")
        userInput, _ = stdin.ReadString('\n')
        userInput = userInput[:len(userInput)-1]

        chatId, err := strconv.ParseInt(userInput, 10, 64)
        
        if err != nil {
            fmt.Fprintf(os.Stderr, "failed to parse '%s' to a number\n", userInput)
        } else {

            fmt.Println("...chat selected...")
            chatLoop := true

            for chatLoop {
                
                fmt.Print("?: ")
                userInput, _ = stdin.ReadString('\n')
                userInput = userInput[:len(userInput)-1]

                //if its a command
                if len(userInput) > 1 {
                    if userInput[0] == '\\' {
                        if len(userInput) > 3 {
                            if userInput[1] == 'v' {
                                sendVoiceMessage(client, chatId, userInput[3:], gLang)
                            } else if userInput[1] == 's' {
                                sendVoiceFile(client, chatId, userInput[3:]) 
                            }
                        } else if userInput[1] == 'q' {
                            chatLoop = false
                        } else if userInput[1] == 'e' { 
                            chatLoop = false 
                            run = false
                        } else {
                            fmt.Println("uknown command")
                        }
                    } else {
                        sendMessage(client, chatId, userInput)
                    }
                } else {
                    sendMessage(client, chatId, userInput)
                } 

            }
        }
    }
}
